dashboard/src/reducers/runtimes.ts

```diff
@@ -0,0 +1,32 @@
+import { getType } from "typesafe-actions";
+
+import actions from "../actions";
+import { RuntimesAction } from "../actions/runtimes";
+import { IRuntime } from "../shared/types";
+
+export interface IRuntimeState {
```

Comment:
as suggested above, I think this could be part of the function reducer and state